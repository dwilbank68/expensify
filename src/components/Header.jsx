import React from 'react';
import PropTypes from 'prop-types';
import {  NavLink} from 'react-router-dom';


// import Header from './Header.jsx';
// const Header = (props) => {
const Header = ({whatever1, whatever2}) => {
    // no lifecycle methods
    // no refs

    const methodName = (e) => {
        //
    }

    return (
        <div>
            <h1>Expensify</h1>
            <NavLink to="/create" activeClassName="is-active">
                create
            </NavLink>
            <NavLink to="/edit" activeClassName="is-active">
                edit
            </NavLink>
            <NavLink to="/help" activeClassName="is-active">
                help
            </NavLink>
            <NavLink to="/" exact={true} activeClassName="is-active">
                home
            </NavLink>
        </div>
    );
};


// Header.defaultProps = {};
// Header.propTypes = {
//     name:        PropTypes.string.isRequired,
//     hndleIptChg: PropTypes.func,
//     id:          PropTypes.oneOfType([ PropTypes.number, PropTypes.string ]).isRequired,
//     message:     PropTypes.shape({ title: PropTypes.string, text: PropTypes.string }).isRequired,
//     comments:    PropTypes.arrayOf(React.PropTypes.object),
//     todos:       PropTypes.array,
//     isComplete:  PropTypes.bool,
//     id:          PropTypes.number,
//     date:        PropTypes.instanceOf(Date)
// };
//
// PropTypes -> array, bool, func, number, object, string, symbol

export default Header;
