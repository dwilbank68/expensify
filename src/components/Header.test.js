import React from 'react';
import {shallow} from 'enzyme';
import {Header} from './Header';

let wrapper, startLogout;

beforeEach(() => {
    startLogout = jest.fn();
    wrapper = shallow(
        <Header startLogout={startLogout}/>
    );
})

test('should render Header', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    wrapper
        .find('button')
        .simulate('click')
    expect(startLogout).toHaveBeenCalled();
})

// .toBe        compare arrays - ensure they are same object, or check numbers
// .toEqual     compare content of arrays, objects