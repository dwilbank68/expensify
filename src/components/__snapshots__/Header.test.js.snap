// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    create
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
